### PR TITLE
Extend the mesh connectivity section

### PR DESCRIPTION
**Pull request by fsimonis**
This PR extends the mesh connectivity section with:
* `setMeshTetrahedron`
* Volume coupling
* explanations of the v3 overhaul

### CODE REVIEW

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md [hunk 1]
```diff
@@ -8,21 +8,27 @@ summary: "So far, our coupling mesh is only a cloud of vertices. This is suffici
 
 The most important example where mesh connectivity is needed is the `nearest-projection` mapping, where the mesh we project _into_ needs mesh connectivity. For a consistent mapping, this is the mesh _from_ which you map. For a conservative mapping, the mesh _to_ which you map. More information is given on the [mapping configuration page](configuration-mapping).
 
-In 2D, mesh connectivity boils down to defining edges between vertices. In 3D, you need to define triangles and / or quads. Both, we can either build up from edges or directly from vertices.
+There are two types of connectivity, which depend on the type of coupling.
+For surface coupling in 2D, mesh connectivity boils down to defining edges between vertices. In 3D, you need to define triangles and / or quads.
+For volume coupling in 2D, mesh connectivity boils down to defining triangles and / or quads between vertices. In 3D, you need to define tetrahedra introduced in version `2.5.0`.
+
+All kind of connectivity can be either build up directly from vertices. Triangles and Quads also allow you to define them using edge IDs.
```
Comment (MakisH) [inline pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md]:
```suggestion
All kind of connectivity can be built up directly from vertices. Triangles and quads also allow us to define them using edge IDs.
```

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md [hunk 2]
```diff
@@ -8,21 +8,27 @@ summary: "So far, our coupling mesh is only a cloud of vertices. This is suffici
...
 void setMeshQuadWithEdges(int meshID, int firstVertexID, int secondVertexID, int thirdVertexID, int fourthVertexID);
+void setMeshTetrahredron(int meshID, int firstVertexID, int secondVertexID, int thirdVertexID, int fourthVertexID);
 ```
 
 * `setMeshEdge` defines a mesh edge between two vertices and returns an edge ID.
 * `setMeshTriangle` defines a mesh triangle by three edges.
 * `setMeshTriangleWithEdges` defines a mesh triangle by three vertices and also creates the edges in preCICE on the fly. Of course, preCICE takes care that no edge is defined twice. Please note that this function is computationally more expensive than `setMeshTriangle`.
 * `setMeshQuad` defines a mesh quad by four edges.
 * `setMeshQuadWithEdges` defines a mesh quad by four vertices and also creates the edges in preCICE on the fly. Again, preCICE takes care that no edge is defined twice. This function is computationally more expensive than `setMeshQuad`.
+* `setMeshTetrahredron` defines a mesh tetrahedra by four vertices.
```
Comment (MakisH) [inline pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md]:
```suggestion
* `setMeshTetrahredron` defines a mesh tetrahedron by four vertices.
```

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md [hunk 3]
```diff
@@ -61,3 +67,30 @@ if(dim==3)
 [...]
 
 ```
+
+## Changes in v3
+
+Version 3 overhauls the definition of meshes.
+
+Connectivity now consists of explicitly defined elements (via the API) and implicitly defined elements.
+preCICE automatically generates all implicit elements.
```
Comment (MakisH) [inline pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md]:
```suggestion
Connectivity now consists of explicitly defined elements (elements created via calls to the API) and implicitly defined elements (elements additionally created by preCICE).
```

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md [hunk 4]
```diff
@@ -61,3 +67,30 @@ if(dim==3)
...
+As an example, explicitly defining a triangle ABC via the API guarantees the existence of the implicit edges AB, AC, and BC.
+
+Furthermore, all connectivity is defined via vertex IDs only. There are no more edge IDs to worry about.
```
Comment (MakisH) [inline pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md]:
```suggestion
Furthermore, all connectivity is defined only via vertex IDs. There are no more edge IDs to worry about.
```

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md [hunk 5]
```diff
@@ -61,3 +67,30 @@ if(dim==3)
...
+The order of vertices also doesn't matter. Triangles BAC and CAB are considered duplicates.
```
Comment (MakisH) [inline pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md]: I assume this is what "considered duplicates" means:

```suggestion
The order of vertices also does not matter. Triangles BAC and CAB are considered duplicates and preCICE will ignore additional definitions of the same triangle.
```

(somehow the `does not` is supposed to be clearer than `doesn't`)

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md [hunk 6]
```diff
@@ -61,3 +67,30 @@ if(dim==3)
...
+
+```cpp
```
Comment (MakisH) [inline pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md:83]: A sentence introducing this list (or an explicit connection to the previous one) is missing.

**Review by MakisH — APPROVED** [review]

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md (hunk 5 above)
Comment (fsimonis) [reply pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md]: The addition is not true. preCICE has to remove duplicates, which is computationally expensive. We should clarify that this comes at a cost.

```suggestion
The order of vertices also does not matter. Triangles BAC and CAB are considered duplicates and preCICE removes one of them during the deduplication step.
```

File: pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md (hunk 6 above)
Comment (fsimonis) [reply pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md:83]:
```suggestion
The API for defining individual connectivity elements looks as follows:

```cpp
```

**Pull request merged**